### ISSUE TITLE
UICollectionViewDataSourcePrefetching seems to be broken

### ISSUE THREAD

**Issue by jspavlick** [arnauddorgans/InfiniteLayout#26]
`func collectionView(_ collectionView: UICollectionView, prefetchItemsAt indexPaths: [IndexPath])` and 
`func collectionView(_ collectionView: UICollectionView, cancelPrefetchingForItemsAt indexPaths: [IndexPath])` are called repeatedly with the same values, making it almost entirely useless.

Has anyone encountered this? Or am I using this library incorrectly?

**Comment by berarduc** [comment]
Yes I see the same problem with very large collections and when scrolling very fast.
I get repeated calls to prefetch / cancel prefetch for the same indexpaths. 
I'm only scrolling in ONE direction, no reversals.
Some indexpaths are included in these repeated calls 2 and 3 times which slows things down for no reason.
